service/matching/version_sets.go

```diff
@@ -30,13 +30,25 @@ import (
 	"crypto/sha256"
 	"encoding/base64"
 
+	commonpb "go.temporal.io/api/common/v1"
 	"go.temporal.io/api/serviceerror"
 	"go.temporal.io/api/taskqueue/v1"
 	"go.temporal.io/api/workflowservice/v1"
 	"go.temporal.io/server/api/persistence/v1"
+	persistencespb "go.temporal.io/server/api/persistence/v1"
 	hlc "go.temporal.io/server/common/clock/hybrid_logical_clock"
 )
 
+var (
+	// TODO: go over error types, maybe not all should be invalid argument
+	errBuildNotFound                     = serviceerror.NewInvalidArgument("build id not found")
+	errNewerBuildFound                   = serviceerror.NewInvalidArgument("newer compatible build exists")
+	errEmptyVersioningData               = serviceerror.NewInvalidArgument("versioning data is empty")
+	errPollWithVersionOnUnversionedQueue = serviceerror.NewInvalidArgument("poll with version capabilities on unversioned queue")
+	errPollOnVersionedQueueWithNoVersion = serviceerror.NewInvalidArgument("poll on versioned queue with no version capabilities")
```

Comment:
I'm assuming this will go away in the next PR when you treat unversioned as "version 0"

Reply:
yes, I'll fix it up later, was just porting old code forward to start